presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/LogicalCteOptimizer.java

```diff
@@ -342,9 +349,29 @@ public boolean isComplexCte(String cteId)
         public List<PlanNode> getTopologicalOrdering()
         {
             ImmutableList.Builder<PlanNode> topSortedCteProducerListBuilder = ImmutableList.builder();
-            Traverser.forGraph(graph).depthFirstPostOrder(graph.nodes())
+            Traverser.forGraph(cteDependencyGraph).depthFirstPostOrder(cteDependencyGraph.nodes())
                     .forEach(cteName -> topSortedCteProducerListBuilder.add(cteProducerMap.get(cteName)));
```

Comment:
Here we are not reversing the list and hence the topological ordering is correct, if we reverse the graph, we should reverse the list here and in SequenceNode